### PR TITLE
Update hypershift dependency to remove replace statements

### PR DESCRIPTION
**Pull request by mjlshen**
Leverage `github.com/openshift/hypershift/api` so that we can remove the `replace` statements that have been carried in go.mod. This will allow us to stop having to pin to old versions of libraries and more easily update dependencies moving forward, namely controller-runtime and K8s!

[OSD-22069](https://issues.redhat.com//browse/OSD-22069)

### CODE REVIEW

**Comment by wanghaoran1988** [comment]
integration tests failed :)

File: go.mod [hunk 1]
```diff
@@ -3,95 +3,78 @@ module github.com/openshift/route-monitor-operator
...
-	k8s.io/apimachinery v0.26.3
-	k8s.io/client-go v12.0.0+incompatible
-	k8s.io/kube-openapi v0.0.0-20230308215209-15aac26d736a
-	sigs.k8s.io/controller-runtime v0.14.4
+	k8s.io/api v0.29.2
+	k8s.io/apiextensions-apiserver v0.29.2
+	k8s.io/apimachinery v0.29.2
+	k8s.io/client-go v0.29.2
+	k8s.io/kube-openapi v0.0.0-20231010175941-2dd684a91f00
+	sigs.k8s.io/controller-runtime v0.17.2
```
Comment (tnierman) [inline go.mod:23]: :raised_hands:

Comment (mjlshen) [reply go.mod:23]: Great question - controller-runtime `v0.18.0` is for K8s `v1.30.z` (and we aren't quite ready for it yet, that's OCP 4.17) https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.18.0

**Comment by mjlshen** [comment]
I don't know why the integration test is failing yet, but when I figure out async, I'll take the PR out of WIP

**Comment by mjlshen** [comment]
Finally figured out the integration test fix 

cc @wanghaoran1988 @tnierman

**Pull request merged**